module/zfs/spa_misc.c

```diff
@@ -2553,6 +2553,26 @@ param_set_deadman_synctime(const char *val, zfs_kernel_param_t *kp)
 	return (0);
 }
 
+static int
+param_set_slop_shift(const char *buf, zfs_kernel_param_t *kp)
```

Comment:
If we're going to go here, and I think we should, then let's make a new parameter to hold the actual result of the MAX/MIN calculation in spa_get_slop_space(). This should save a CPU cycle or two since the MAX/MIN is re-evaluated every txg sync in metaslab_sync_done().

Reply:
"_a new parameter_" as in "_a new kernel module parameter_"? I'm not exactly thrilled by the idea of adding a new "parameter" which should really be a user-configurable tunable but here we use it as a "cached result" just to save a couple of CPU cycles: it seems confusing from a usability standpoint.

Reply:
We should consider converting this to a pool property.  This would allow it to default to a reasonable value on a per-pool basis, which is useful when a single system has both large and small pools.  Making it persistent also avoids some upgrade issues we've seen historically where the default system value changes and a pool's free space is suddenly very different.

That said, there's no need to make that change in this PR, unless you're feeling ambitious!  But if we agree that's the right thing to do, then I don't think we want to over optimize this and it's best to keep this change minimal and simply address the reported issue.

Reply:
@behlendorf i think it's a great idea, and i am going to try and implement it after i am done teaching zed to offline physically removed spares (if i succeed).

Reply:
That's great.  Then why don't we plan to go ahead and merge this as-is, which will make it relatively easy to cherry-pick to 0.7 if needed.  Then you can follow up with a new PR which is a property based version.

Reply:
@behlendorf sure, i will open a new PR for review as soon as it is implemented.